sdk/core/azure-core/src/http/response.cpp

```diff
@@ -40,18 +38,72 @@ inline bool IsStringEqualsIgnoreCase(std::string const& a, std::string const& b)
 }
 } // namespace
 
-void Response::AddHeader(std::string const& name, std::string const& value)
+void Response::AddHeader(std::string const& header)
 {
-  if (IsStringEqualsIgnoreCase("Content-Length", name))
+  // get name and value from header
+  auto start = header.begin();
+  auto end = std::find(start, header.end(), ':');
+
+  if (end == header.end())
   {
-    // whenever this header is found, we reserve the value of it to be pre-allocated to write
-    // response
-    m_bodyBuffer.reserve(std::stol(value));
+    return; // not a valid header or end of headers symbol reached
   }
+
+  auto headerName = std::string(start, end);
+  start = end + 1; // start value
+  while (start < header.end() && (*start == ' ' || *start == '\t'))
+  {
+    ++start;
+  }
+
+  end = std::find(start, header.end(), '\r');
+  auto headerValue = std::string(start, end); // remove \r
+
+  AddHeader(headerName, headerValue);
+}
+
+void Response::AddHeader(std::string const& name, std::string const& value)
+{
+  /* if (IsStringEqualsIgnoreCase("Content-Length", name))
+  {
+    if (this->m_bodyStream != nullptr)
+    {
+      // Create an empty stream just to hold the size. Then any transport can replace for an
+      // specific stream, like libcurl, it will get the size and creates a CurlBodyStream
+      SetBodyStream(new MemoryBodyStream(nullptr, std::stol(value)));
+    }
+    else
+    {
+      // whenever this header is found, we reserve the value of it to be pre-allocated to write
+      // response
+      m_bodyBuffer.reserve(std::stol(value));
+    }
+  } */
   this->m_headers.insert(std::pair<std::string, std::string>(name, value));
 }
 
-void Response::AppendBody(uint8_t* ptr, uint64_t size)
+void Response::SetBodyStream(BodyStream* stream)
+{
+  // Before setting body Stream, avoid leaking
+  if (this->m_bodyStream != nullptr)
+  {
+    delete this->m_bodyStream;
+  }
+  this->m_bodyStream = stream;
+}
+
+std::unique_ptr<std::vector<uint8_t>> Response::GetBodyBufferFromStream(BodyStream* const stream)
```

Comment:
The name is misleading. `Get` implicate that this only gets an existing buffer, however, this function is actually costly and heavy. `Create`/`Construct` should be better.

Reply:
update it to construct...
I will still ask @JeffreyRichter  about if we want to offer this function in the HTTP::Response or not.
I added as a way to simplify all the places where storage is currently expecting a bodyBuffer from the Response.